cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -509,9 +509,15 @@ private MetadataEntry convertRow(Row row) {
     byte[] keyBytes = mdsKey.getKey();
     // byte array is automatically initialized to 0, which implies fixed match in fuzzy info
     // the row key after targetId doesn't need to be a match.
-    byte[] infoBytes = new byte[keyBytes.length];
-
-    return new ImmutablePair<>(keyBytes, infoBytes);
+    // From HBase 1.1 the HBase FuzzyRowFilter has a pre-processing step for the mask and before pre-procesing it checks
+    // the mask array indices to atleast have one value which is not equal to -1 and not equal to 0. So to the
+    // infoByte (mask) append 1 in the last to to fail this mask and trigger the pre-processing. For more details:
+    // please see org.apache.hadoop.hbase.filter.FuzzyRowFilter constructor
```

Comment:
We can just say -
Workaround for HBASE-15676, need to have at least one `1` in the  fuzzy filter.